pages/GSoC-21.md

```diff
@@ -0,0 +1,156 @@
+- Working on open issues is the best way for your skill to be recognised.
+- It is a good practice to search the history of issues for similar problems before creating a new issue on GitHub, this will avoid the creation of duplicated issues.
+- Try running our tools and going through the demos to get a better understanding of how things work.
+- Hang around on the Slack channel and talk to people about any problems or issues.
+- Make sure that any issue your create is verified before beginning to work on it.
+- Use the provided [template](https://docs.google.com/document/d/1qPR02o6jY4uFBdCf3S0ous4LG32YCe6ZJuDGojXH2JY/edit?usp=sharing) for writing your proposals.
+- Discuss your ideas with the mentors and get your proposals reviewed before submitting. [Note: Due to the large number of proposals we receive, we only review a proposal once per student, so make it count.]
+- A list of prerequisites and concepts needed to understand Hydra and the tools developed by Hydra Ecosystem can be found [here](https://www.hydraecosystem.org/Starting-Material).
+- If you have any doubts, ask questions in our super friendly Slack community. We'll try our best to answer all your questions.
+
+<div id="Ideas"></div>
+
+# [Ideas for GSoC 2021](#Ideas)
+
+## 1. General Improvements in [hydrus](https://github.com/HTTP-APIs/hydrus).
+### Key points:
+There is always room from improvement in our flagship server, hydrus:
+- Updating the server according to the Hydra Spec.
+- Creation of dynamic endpoints. Users need to be able to define endpoints that have some functionality. Right now hydrus can only serve static data and users are unable to define how incoming data is processed in a hydrus server.
+- General improvements in code and optimizing the server functionality, refactoring.
+- Keep up with the development of the querying system for the agent
+- Implement generalisation of Collection as a Resource.
+- Add more robust security features to hydrus.
+
+**Skills Required**:
+- Python
+- Hydra
+- Semantic Web
+- Ability to write test suites
+- Software Engineering
+
+**Difficulty Level**: *Intermediate*
+
+**Issues**<br/>
+Some issues are already open for these, please check [issues](https://github.com/HTTP-APIs/hydrus/issues).
+
+
+## 2. General Improvements in [agent](https://github.com/HTTP-APIs/hydra-python-agent).
+### Key points:
+The agent is a super cool way to interact with a hydra server, but there is lot of room for improvement. Some parts of this project can be:
+- Improving the UI/UX of the agent.
+- Add graph interaction events/actions support.
+- Making the agent more responsive.
+- Generalising the agent so that it works with any hydra enabled API. Right now the agent is strongly tied up with hydrus.
+- Implement sync-mechanism for non-hydrus Hydra servers. (No backend updates pushed from the server)
+- Optimise the graph storage and search mechanism in the client.
+- Implement better design patterns for the frontend code.
+
```

Comment:
This one should be fine IMO. @sameshl please modify/add things you want to get done.

Reply:
Looks fine to me. @priyanshunayan can give more details

Reply:
Things that I think should be there in GUI:
1. Bring on par with hydrus and hydra-python-agent.
2. Display object instances in the graph too realtime. 
3. Deploy with docker container so that other people can use it on the web without having to install it locally.